launch/launch/__init__.py

```diff
@@ -75,8 +78,11 @@ def __init__(self, cmd, name, output_handler, exit_handler, env=None):
 
     def send_signal(self, signal):
         if self.transport:
+            self.task_state.signals_received.append(signal)
             self.transport.send_signal(signal)
 
     def terminate(self):
-        if self.transport:
+        if os.name != 'nt':
+            self.send_signal(signal.SIGINT)
```

Comment:
How does manually overriding the `exit_future` ensure that the process is terminated? It will "signal" like it has been terminated but the process might still be running, no?